doc/tutorials/using_zephyr_as_uos.rst

```diff
@@ -0,0 +1,88 @@
+.. _using_zephyr_as_uos:
+
+Using Zephyr as User OS
+#######################
+
+The tutorial describes the steps to run zephyr as User OS on ACRN hypervisor. We are using
+Kaby Lake-based NUC model NUC7i5DNH and you can try other platform ACRN supported as well.
+
+Introduction to Zephyr
+**********************
+
+The Zephyr is a scalable real-time operating-system (RTOS) supporting multiple hardware architecture,
+optimized for resource constrained devices, and built with safety and security in mind.
+
+Steps for Using Zephyr as User OS
+*********************************
+
+#. Build Zephyr and grub2 boot loader image
+
+   First of all, please follow the `Development Environment Setup on Linux
+   <https://docs.zephyrproject.org/latest/getting_started/installation_linux.html>`_ to
+   setup Zephyr development environment. Then follow the `Board-Specfic Support
+   <https://docs.zephyrproject.org/latest/boards/index.html>`_ to build the acrn supported zephyr image ``zephyr.elf``.
+
+   Build grub2 boot loader image
+
+   .. code-block:: none
+
+      $ git clone https://github.com/zephyrproject-rtos/zephyr.git
+      $ cd $ZEPHYR_BASE
+      $ ./boards/x86/common/scripts/build_grub.sh x86_64
+
+   Find the binary at boards/x86/common/scripts/grub/bin/grub_x86_64.efi.
+
+#. Preparing the boot device
+
+   .. code-block:: none
+
+      $ dd if=/dev/zero of=zephyr.img bs=1M count=1024
+      $ mkfs.vfat -F 32 zephyr.img
+      $ losetup -f -P --show zephyr.img
+      $ mount /dev/loop0 /mnt
+
+   Create the following directories.
+
+   :file:`/mnt/efi`
+
+   :file:`/mnt/efi/boot`
+
+   :file:`/mnt/kernel/`
+
+   Copy ``zephyr.elf`` to ``/mnt/kernel/`` and ``grub_x86_64.efi`` to ``/mnt/efi/boot/``.
+
+   Create ``/mnt/efi/boot/grub.cfg`` containing the following:
+
+   .. code-block:: console
+
+     set default=0
+     set timeout=10
+
+     menuentry "Zephyr Kernel" {
+         multiboot /kernel/zephyr.elf
+     }
+
+#. Follow `Getting started guide for Intel NUC <https://projectacrn.github.io/latest/getting-started/apl-nuc.html>`_ to
+   boot "The ACRN Service OS"
+
+#. Boot Zephyr as User OS
+
+   Copy the launch script ``devicemodel/samples/nuc/launch_zephyr.sh``, ``devicemodel/bios/OVMF.fd`` and ``zephyr.img``
```

Comment:
This kind of implies the user did a `git clone ...` for the ACRN repo but that's not in this guide. Assuming the user has followed the ACRN GSG steps he should actually have the `launch_zephyr.sh` scripts on his filesystem already, same for the `OVMF.fd` file so we should simply instruct the user to pick these up from `/usr/share/acrn/samples/nuc/` and `/usr/share/acrn/samples/bios/` (going from memory for the exact paths, so to be checked).

Reply:
Yes. It is a better way. I will update it.